aspnet/mobile/native-mobile-backend.rst

```diff
@@ -1,11 +1,170 @@
+  :language: c#
+
+For this sample, the implementation just uses a private collection of items. Configure the implementation in `Startup.cs`:
+
+.. literalinclude:: native-mobile-backend/sample/ToDoApi/src/ToDoApi/Startup.cs
+  :language: c#
+  :lines: 30-36
+  :dedent: 8
+  :emphasize-lines: 6
+  
+At this point, you're ready to create the `TodoItemsController`.
+
+Creating the Controller
+-----------------------
+
+Add a new controller to the project, `TodoItemsController`. It should inherit from Microsoft.AspNetCore.Mvc.Controller. Add a ``Route`` attribute to indicate that the controller will handle requests made to paths starting with ``api/todoitems``, which may optionally include an ``id`` argument as part of the URL path. The ``[controller]`` token in the route is replaced by the name of the controller (omitting the ``Controller`` suffix), and is especially helpful for global routes. Learn more about :doc:`routing </fundamentals/routing>`.
+
+The controller requires an ``ITodoRepository`` to function; request an instance of this type through the controller's constructor. At runtime, this instance will be provided using the framework's support for :doc:`dependency injection </fundamentals/dependency-injection>`.
+
+.. literalinclude:: native-mobile-backend/sample/ToDoApi/src/ToDoApi/Controllers/TodoItemsController.cs
+  :language: c#
+  :lines: 1-17
+
+This API supports four different HTTP verbs to perform CRUD (Create, Read, Update, Delete) operations on the data source. The simplest of these is the Read operation, which corresponds to an HTTP GET request.
+
+Reading Items
+^^^^^^^^^^^^^
+
+Requesting a list of items is done with a GET request to the List() method. The method name doesn't matter as long as it's the only method matching the request for the given HTTP verb (in this case, GET). If there were multiple GET methods, it would be necessary to add ``[Route]`` attributes to each action (or add ``[action]`` to the route, which would use the action's name as part of the route).
+
+.. literalinclude:: native-mobile-backend/sample/ToDoApi/src/ToDoApi/Controllers/TodoItemsController.cs
+  :language: c#
+  :lines: 19-23
+  :dedent: 8
+
+You can test your new API method using a variety of tools, such as `Postman <https://www.getpostman.com/docs/>`_, shown here:
+
+.. image:: native-mobile-backend/_static/postman-get.png
+
+Creating Items
+^^^^^^^^^^^^^^
+
+By convention, creating new data items is mapped to the HTTP POST verb. The ``Create`` method has an ``[HttpPost]`` attribute applied to it, and accepts an ID parameter and a ``TodoItem`` instance. Since the item will be passed in the body of the POST, this parameter is decorated with the ``[FromBody]`` attribute.
+
+Inside the method, the item is checked for validity and prior existence in the data store, and if no issues occur, it is added using the repository. The error handling  mirrors the behavior of the original service used by the native app.
```

Comment:
> The error handling  mirrors the behavior of the original service used by the native app.

I think it would be good to explicitly describe what `ModelState.IsValid` indicates. We shouldn't assume the reader is familiar with the original sample or with ASP.NET Web API.